include/benchmark/benchmark.h

```diff
@@ -378,6 +378,18 @@ enum BigO { oNone, o1, oN, oNSquared, oNCubed, oLogN, oNLogN, oAuto, oLambda };
 // computational complexity for the benchmark.
 typedef double(BigOFunc)(int);
 
+// StatisticsFunc is passed to a benchmark in order to compute some descriptive
+// statistics over all the measurements of some type
+typedef double(StatisticsFunc)(const std::vector<double>&);
+
+struct Statistics {
```

Comment:
i think this can be a fwd decl here with the implementation in the `benchmark.cc` file. this helps discourage library users from trying to instantiate one of these, which given the API below they shouldn't need to.